src/Arcus.EventGrid.Tests.Integration/Publishing/EventPublishingTests.cs

```diff
@@ -181,19 +206,94 @@ await EventGridPublisherBuilder
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            const string licensePlate = "1-TOM-1337";
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: $"{{\"licensePlate\": \"{licensePlate}\"}}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyAsync(events);
+
+            // Assert
+            Assert.All(events, rawEvent => AssertReceivedEvent(rawEvent, licensePlate, GetReceivedEvent.WithRetryCount));
+        }
+
+        [Fact]
+        public async Task PublishMultipleRawEvents_WithBuilder_ValidParameters_SucceedsWithTimeout()
+        {
+            // Arrange
+            var topicEndpoint = Configuration.GetValue<string>("Arcus:EventGrid:TopicEndpoint");
+            var endpointKey = Configuration.GetValue<string>("Arcus:EventGrid:EndpointKey");
+            const string licensePlate = "1-TOM-1337";
+            var events =
+                Enumerable
+                    .Repeat<Func<Guid>>(Guid.NewGuid, 2)
+                    .Select(newGuid => new RawEvent(
+                                newGuid().ToString(),
+                                subject: "integration-test",
+                                body: $"{{\"licensePlate\": \"{licensePlate}\"}}",
+                                type: "Arcus.Samples.Cars.NewCarRegistered",
+                                dataVersion:"1.0",
+                                eventTime: DateTimeOffset.Now))
+                    .ToArray();
+
+            // Act
+            await EventGridPublisherBuilder
+                  .ForTopic(topicEndpoint)
+                  .UsingAuthenticationKey(endpointKey)
+                  .Build()
+                  .PublishManyRawAsync(events);
+
+            // Assert
+            Assert.All(events, rawEvent => AssertReceivedEvent(rawEvent, licensePlate, GetReceivedEvent.WithTimeout));
+        }
+
+        private enum GetReceivedEvent { WithRetryCount, WithTimeout }
+
+        private void AssertReceivedEvent(IEvent @event, string licensePlate, GetReceivedEvent getReceivedEvent)
+        {
+            TracePublishedEvent(@event.Id, @event);
+            string GetReceivedEvent()
+            {
+                switch (getReceivedEvent)
```

Comment:
Instead of using a switch it's better to have retry count and timeout as optional parameters on the method.

Reply:
Not sure what that would do, since the two calls on the `EventConsumerHost` are totally different and having both of them inside the method would make the check to know which one to use unnecessary complex.

Reply:
In that case I'd say they are 2 different scenarios and thus different methods